File: backend/models/Vehicule.py
from django.db import models

class Vehicule(models.Model):
    id = models.AutoField(primary_key=True)
    immatriculation = models.CharField(max_length=50)
    modele = models.CharField(max_length=50)
    marque = models.CharField(max_length=30)
    couleur = models.CharField(max_length=30)
    nb_place = models.IntegerField()

File: frontend/src/app/models/vehicule.ts
import { Site } from "./site";

export class Vehicule{
    public Id: number;
    public Immatriculation: string;
    public Modele: string;
    public Marque: string;
    public Couleur: string;
    public NbPlace: string;
    public IsActive: boolean;
    public Site: Site;

    public constructor(init?: Partial<Vehicule>) {
        Object.assign(this, init);
    }
}
